src/test/java/io/fabric8/maven/docker/config/RunImageConfigurationTest.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.fabric8.maven.docker.config;
+
+import org.apache.commons.lang3.SerializationUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.*;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class RunImageConfigurationTest {
+    private RunImageConfiguration runImageConfiguration;
+
+    @Test
+    public void testEnv() {
+        Map<String, String> envMap = new HashMap<>();
+
+        envMap.put("ADDR_TYPE_1", "${docker.container.consul.ip}:8500");
+        envMap.put("ADDR_TYPE_2", "+${docker.container.consul.ip}");
+
+        runImageConfiguration = new RunImageConfiguration.Builder()
+                .env(envMap)
+                .build();
+
+        Assert.assertNotNull(runImageConfiguration);
+        Assert.assertNotNull(runImageConfiguration.getEnv());
+        Assert.assertEquals("${docker.container.consul.ip}:8500", runImageConfiguration.getEnv().get("ADDR_TYPE_1"));
+        Assert.assertEquals("+${docker.container.consul.ip}", runImageConfiguration.getEnv().get("ADDR_TYPE_2"));
+    }
+
+    @Test
+    public void testNetworkingConfig() {
+        runImageConfiguration = new RunImageConfiguration.Builder()
+                .network(null)
+                .build();
+        Assert.assertNotNull(runImageConfiguration);
+        Assert.assertNotNull(runImageConfiguration.getNetworkingConfig());
+    }
+
+    @Test
+    public void testSerialization() {
```

Comment:
what is this test about ?

Reply:
Sorry, I thought I should add a test to check whether these RunImageConfiguration objects are properly serialized into objects or not. If this doesn't look like a sane to you, I would remove it.

Also, Can there be other tests related to this class?

Reply:
well, don't know exactly where this class should be serialized. 

Adding tests are very valuable, but:

* they add maintenance costs, which is ok if the test has some value and tests the right thing. E.g they need to be updated if ImageConfiguration changes
* should not test trivial things which is out of scope of the project. E.g. testing Java serialization (which is the task of the JVM) is not in scope of this project. The same btw for testing setter and getter.

So, testing simple, dump configuration classes does not make sense normally, except when it add some extra logic.

What should be tested for sure are classes which _use_ these configuration. 

E.g. the latest PR about allowing to merge property based configuration with XML based configuration introduces some kind of logic to the configuration classes which could (and probably should) be tested. However, the plan is to move these classes to a `Optional` to handle the situation better to distinguish between default values and values set by the user when it comes to mergign (where default values should be ignored).

tl;dr - yes, please remove this test ;-) (but thanks for caring about testing, not every contribution does this)

Reply:
Sorry for my naive mistakes. I'll update it

Reply:
hey @rohanKanojia no need to apologize ! it's awesome that you picked up the story and did the implementation. Highly appreciated !